### ISSUE TITLE
There seems to be an issue with scalacOptions being applied for ScalaJs and ScalaNative test projects

### ISSUE THREAD

**Issue by DamianReeves** [lolgab/mill-crossplatform#9]
The parent's `scalacOptions` are not being carried over for ScalaJs and ScalaNative test projects.

I added some tests for the problem in the `itest` folder here: https://github.com/DamianReeves/mill-crossplatform/tree/with-testing-and-nesting

**Comment by lolgab** [comment]
I think this is a Mill bug, not a mill-crossplatform one.
Repro:
```scala
import mill.scalalib._
import mill.scalajslib._

object module extends ScalaJSModule {
  def scalaVersion = "2.13.10"
  def scalaJSVersion = "1.12.0"
  def scalacOptions = super.scalacOptions() ++ Seq("-deprecation")

  object test extends Tests with TestModule.Utest
}
```

```
↪ mill show module.scalacOptions
Compiling /Users/lorenzo/scala/example-mill-4/build.sc
[1/1] show 
[1/1] show > [2/2] module.scalacOptions 
[
  "-deprecation"
]
lorenzo at MacBook-Pro-di-Lorenzo in ~/s/example-mill-4
↪ mill show module.test.scalacOptions
[1/1] show 
[1/1] show > [1/1] module.test.scalacOptions 
[
  
]
```

**Comment by DamianReeves** [comment]
D'uh ... spent all this effort to reproduce this and I didn't try the simplest thing.

**Comment by lolgab** [comment]
To be fixed in Mill. PR here: https://github.com/com-lihaoyi/mill/pull/2274

**Comment by lolgab** [comment]
This is fixed in Mill `0.11.0+`
